src/server.c

```diff
@@ -3461,56 +3461,58 @@ int processCommand(client *c) {
         return C_ERR;
     }
 
-    /* Set server.script_oom to 0, if we will find out that we are
-     * over the memory limit, it will be set to 1. */
-    server.script_oom = 0;
-
     /* Handle the maxmemory directive.
      *
      * Note that we do not want to reclaim memory if we are here re-entering
      * the event loop since there is a busy Lua script running in timeout
      * condition, to avoid mixing the propagation of scripts with the
      * propagation of DELs due to eviction. */
-    if (server.maxmemory && !scriptIsTimedout()) {
-        int out_of_memory = (performEvictions() == EVICT_FAIL);
-
-        /* performEvictions may evict keys, so we need flush pending tracking
-         * invalidation keys. If we don't do this, we may get an invalidation
-         * message after we perform operation on the key, where in fact this
-         * message belongs to the old value of the key before it gets evicted.*/
-        trackingHandlePendingKeyInvalidations();
-
-        /* performEvictions may flush slave output buffers. This may result
-         * in a slave, that may be the active client, to be freed. */
-        if (server.current_client == NULL) return C_ERR;
-
-        int reject_cmd_on_oom = is_denyoom_command;
-        /* If client is in MULTI/EXEC context, queuing may consume an unlimited
-         * amount of memory, so we want to stop that.
-         * However, we never want to reject DISCARD, or even EXEC (unless it
-         * contains denied commands, in which case is_denyoom_command is already
-         * set. */
-        if (c->flags & CLIENT_MULTI &&
-            c->cmd->proc != execCommand &&
-            c->cmd->proc != discardCommand &&
-            c->cmd->proc != quitCommand &&
-            c->cmd->proc != resetCommand) {
-            reject_cmd_on_oom = 1;
-        }
+    if (!scriptIsTimedout()) {
+        /* Set server.script_oom to 0, if we will find out that we are
+         * over the memory limit, it will be set to 1. */
+        server.script_oom = 0;
+        if (server.maxmemory) {
+            int out_of_memory = (performEvictions() == EVICT_FAIL);
```

Comment:
i'm still not sure what we're trying to solve. is it so that we can avoid testing `server.maxmemory` in the scripting c files?
i'd rather find a way to do that without the massive indentation change. maybe using an additional `if` or `else-if`

Reply:
I am trying to solve the following:
```
127.0.0.1:6379> FUNCTION load lua test "redis.register_function('f1', function() return 1 end)"
OK
127.0.0.1:6379> fcall f1 0
(integer) 1
127.0.0.1:6379> config set maxmemory 1
OK
127.0.0.1:6379> fcall f1 0
(error) OOM allow-oom flag is not set on the function, can not run it when used memory > 'maxmemory'
127.0.0.1:6379> config set maxmemory 0
OK
127.0.0.1:6379> fcall f1 0
(error) OOM allow-oom flag is not set on the function, can not run it when used memory > 'maxmemory'
```
I can also check `server.maxmemory` on `fcallCommand`, will change it.

Reply:
ohh, is there such a bug in 6.2 too? (in which case i'd appreciate a PR against that branch)

Reply:
No when we checked for maxmemory on `redis.call` we check if `server.maxmemory` is set and if not we are not even checking `server.script_oom`. This is only related to this PR where we do not even start to run the function if we reached OOM. I can do the same trick here.

Reply:
Fixed